Docs/user_guide/citing.rst

```diff
@@ -0,0 +1,46 @@
+======
+Citing
+======
+
+3D Slicer as a Platform
+-----------------------
+
+To acknowledge 3D Slicer as a platform, please cite the `Slicer web site <http://www.slicer.org/>`_ and the following publications when publishing work that uses or incorporates 3D Slicer: 
+
+
+Slicer 4
+^^^^^^^^
+
+Fedorov A., Beichel R., Kalpathy-Cramer J., Finet J., Fillion-Robin J-C., Pujol S., Bauer C., Jennings D., Fennessy F.M., Sonka M., Buatti J., Aylward S.R., Miller J.V., Pieper S., Kikinis R. `3D Slicer as an Image Computing Platform for the Quantitative Imaging Network <http://www.spl.harvard.edu/publications/item/view/2219>`_. Magn Reson Imaging. 2012 Nov;30(9):1323-41. PMID: 22770690. PMCID: PMC3466397.
+
+
+Slicer 3
+^^^^^^^^
+
+Pieper S, Lorensen B, Schroeder W, Kikinis R. `The NA-MIC Kit: ITK, VTK, Pipelines, Grids and 3D Slicer as an Open Platform for the Medical Image Computing Community <http://www.spl.harvard.edu/publications/item/view/68>`_. Proceedings of the 3rd IEEE International Symposium on Biomedical Imaging: From Nano to Macro 2006; 1:698-701.
+
+Pieper S, Halle M, Kikinis R. `3D SLICER <http://www.spl.harvard.edu/publications/item/view/91>`_. Proceedings of the 1st IEEE International Symposium on Biomedical Imaging: From Nano to Macro 2004; 1:632-635.
+
+
+Slicer 2
+^^^^^^^^
+
+Gering D.T., Nabavi A., Kikinis R., Hata N., O'Donnell L., Grimson W.E.L., Jolesz F.A., Black P.M., Wells III W.M. An `Integrated Visualization System for Surgical Planning and Guidance using Image Fusion and an Open MR <http://www.spl.harvard.edu/publications/item/view/156>`_. J Magn Reson Imaging. 2001 Jun;13(6):967-75. PMID: 11382961.
+
+Gering D.T., Nabavi A., Kikinis R., Grimson W.E.L., Hata N., Everett P., Jolesz F.A., Wells III W.M. `An Integrated Visualization System for Surgical Planning and Guidance using Image Fusion and Interventional Imaging <http://www.spl.harvard.edu/publications/item/view/816>`_. Int Conf Med Image Comput Comput Assist Interv. 1999 Sep;2:809-19. 
+
+
+Individual Module
+-----------------
+
+To acknowledge individual modules:
+
+.. figure:: images/Acknowledgement-tab.png
```

Comment:
We should probably move all images to https://discourse.slicer.org/c/dev/documentation-resources/26 and not upload these images here.

Reply:
Good point, I totally forgot about this. I will do so.

Reply:
But first, going on a :running_man: before the :sun_with_face: set

Reply:
When you first drag and drop an image into  a discourse post, it is uploaded into a temporary location and is available using an url like this one (available doing right click `Copy Image Location` in the preview):

```
https://aws1.discourse-cdn.com/standard17/uploads/slicer/original/2X/d/sywbxlReSbohAlm7G5Hy8s47Osp.png
```
and it generates markdown like this one:

```markdown
![auto_complete|21x21](upload://sywbxlReSbohAlm7G5Hy8s47Osp.png)
```

Then, after saving the post, the markdown remains unchanged but the image is moved to an amazon S3 bucket and the url become:

``'
https://discourse-cloud-file-uploads.s3.dualstack.us-west-2.amazonaws.com/standard17/uploads/slicer/original/2X/c/c823cbb536a39bef882e6b75d10c3b9e8e92d931.png
```

where `discourse-cloud-file-uploads` is a bucket and `standard17/uploads/slicer/original/2X/c/c823cbb536a39bef882e6b75d10c3b9e8e92d931.png` is the key

Reply:
Additionally, discourse server post-process the image to decrease it size (in this example from 316 bytes to 208 bytes) and the S3 key likely includes the SHA1 of the post-processed file (`c823cbb536a39bef882e6b75d10c3b9e8e92d931` in our example)

This means that if discourse backend switch to a different optimization of upload images  and re-process all the images ... all the link we use in readthedocs will be broken.

Reply:
Looking at the code, it turns out that only `png` file are optimized:

https://github.com/discourse/discourse/blob/efd5fb665b0f478c0159b7dfb315dc0afa820c17/lib/upload_creator.rb#L334-L342

Reply:
I am now able to recover the SHA1 by running `optipng`:

```
$ src/optipng/optipng -o2 /path/to/Slicer/Docs/user_guide/images/module_segment_editor/auto_complete.png -keep -strip all

$ sha1sum /path/to/Slicer/Docs/user_guide/images/module_segment_editor/auto_complete.png*
c823cbb536a39bef882e6b75d10c3b9e8e92d931  /path/to/Slicer/Docs/user_guide/images/module_segment_editor/auto_complete.png
93070ab5f8ecf49ed4c01a98936834ea3b1762e8  /path/to/Slicer/Docs/user_guide/images/module_segment_editor/auto_complete.png.bak
```

`c823cbb536a39bef882e6b75d10c3b9e8e92d931` is the SHA1 found in the URL

Reply:
Thanks for investigating this, @jcfr ! So the bottom line is that we can use discourse as backend even after they decide to regenerate the optimized images?

Reply:
> So the bottom line is that we can use discourse as backend even after they decide to regenerate the optimized images?

Indeed, now we understand the underlying process to generate the S3 keys, I am more confident to move forward with this approach. 

I also wrote a python scripts allowing to easily generate tables like the one displayed in this post:

https://discourse.slicer.org/t/segment-editor-module-resources/10841